components/inputs/test/input-checkbox.test.js

```diff
@@ -225,23 +156,92 @@ describe('d2l-input-checkbox', () => {
 
 	});
 
-	describe('focus management', () => {
+	describe('property binding', () => {
 
 		let elem;
 		beforeEach(async() => {
 			elem = await fixture(uncheckedFixture);
 		});
 
-		it('should fire "focus" event when input element is focussed', async() => {
-			setTimeout(() => getInput(elem).focus());
-			const { target } = await oneEvent(elem, 'focus');
-			expect(target).to.equal(elem);
+		it('should bind "value" attribute to input property', async() => {
+			elem.setAttribute('value', 'hello');
+			await elem.updateComplete;
+			expect(getInput(elem).value).to.equal('hello');
 		});
 
-		it('should fire "focus" event when custom element is focussed', async() => {
-			setTimeout(() => elem.focus());
-			const { target } = await oneEvent(elem, 'focus');
-			expect(target).to.equal(elem);
+		it('should bind "indeterminate" attribute to input property', async() => {
+			elem.setAttribute('indeterminate', 'indeterminate');
+			await elem.updateComplete;
+			expect(getInput(elem).indeterminate).to.be.true;
+		});
+
+		it('should bind "checked" attribute to input property', async() => {
+			elem.setAttribute('checked', 'checked');
+			await elem.updateComplete;
+			expect(getInput(elem).checked).to.be.true;
+		});
+
+		it('should bind "checked" property to input property', async() => {
+			elem.checked = true;
+			await elem.updateComplete;
+			expect(getInput(elem).checked).to.be.true;
+		});
+
+		it('should continue to bind "checked" property after interaction', async() => {
+			const input = getInput(elem);
+			input.click();
+			expect(input.checked).to.be.true;
+			expect(elem.checked).to.be.true;
+			await elem.updateComplete;
+			elem.checked = false; // eslint-disable-line require-atomic-updates
+			await elem.updateComplete;
+			expect(input.checked).to.be.false;
+		});
+
+		[
+			{name: 'aria-label', propName: 'ariaLabel', value: 'hello'},
+			{name: 'disabled', value: true},
+			{name: 'name', value: 'jim'}
+		].forEach((attr) => {
+			attr.propName = attr.propName || attr.name;
+			it(`should bind "${attr.name}" to input`, async() => {
+				if (typeof attr.value === 'boolean') {
+					elem.setAttribute(attr.name, attr.name);
+				} else {
+					elem.setAttribute(attr.name, attr.value);
+				}
+				await elem.updateComplete;
+				const input = getInput(elem);
+				if (typeof attr.value === 'boolean') {
+					expect(input.hasAttribute(attr.name)).to.be.true;
+					expect(input[attr.propName]).to.equal(attr.value);
+				} else {
+					expect(input.getAttribute(attr.name)).to.equal(attr.value.toString());
+					if (attr.name.indexOf('aria-') === -1) {
+						expect(input[attr.propName]).to.equal(attr.value);
+					}
+				}
+			});
+		});
+
+	});
+
+	describe('simulateClick', () => {
```

Comment:
nit: seems like you could just have the `should trigger an event` test since it checks for `checked` anyways.

Reply:
Yep -- simplified it, thx.